includes/embeds/class-amp-core-block-handler.php

```diff
@@ -395,19 +386,19 @@ static function ( $matches ) use ( $modal_state_property, $is_interactive_block
 				}
 
 				if ( $is_interactive_block ) {
-					// Replace `data-wp-on--click` with AMP state on submenu open button.
+					// Replace `data-wp-on--click` or `data-wp-on-async--click` with AMP state on submenu open button.
 					if ( false !== strpos( $new_block_content, 'wp-block-navigation__responsive-container-open' ) ) {
 						$new_block_content = preg_replace(
-							'/\sdata-wp-on--click="[^"]+"/',
+							'/\sdata-wp-on-(?:-click|async--click)="[^"]+"/',
 							sprintf( ' on="tap:AMP.setState({ %1$s: !%1$s })"', esc_attr( $modal_state_property ) ),
 							$new_block_content
 						);
 					}
 
-					// Replace `data-wp-on--click` with AMP state on submenu close button.
+					// Replace `data-wp-on--click` or `data-wp-on-async--click` with AMP state on submenu close button.
 					if ( false !== strpos( $new_block_content, 'wp-block-navigation__responsive-container-close' ) ) {
 						$new_block_content = preg_replace(
-							'/\sdata-wp-on--click="[^"]+"/',
+							'/\sdata-wp-on-(?:-click|async--click)="[^"]+"/',
```

Comment:
Slightly less redundant:

```suggestion
							'/\sdata-wp-on(?:-async)?--click="[^"]+"/',
```